baseline/baseline.py

```diff
@@ -82,79 +105,169 @@ def load_model(model_file_path: str, device: str = "cpu") -> Any:
+    if audio.ndim != 2:
+        raise ValueError(
+            f"audio input tensor must be 2D with shape (batch_size, num_samples)"
+        )
+
+    # Make sure the correct model type was passed in
+    if not isinstance(model, RandomProjectionMelEmbedding):
+        raise ValueError(
+            f"Model must be an instance of {RandomProjectionMelEmbedding.__name__}"
+        )
+
+    # Send the model to the same device that the audio tensor is on.
     model = model.to(audio.device)
 
-    # Implement batching of the audio
-    if batch_size is None:
-        # Here we just pick a sensible default batch size
-        batch_size = 512
-    dataset = torch.utils.data.TensorDataset(audio)
+    # Split the input audio signals into frames and then flatten to create a tensor
+    # of audio frames that can be batch processed. We will unflatten back out to
+    # (audio_baches, num_frames, embedding_size) after creating embeddings.
+    frames = frame_audio(audio, model.n_fft, hop_size, center)
+    audio_batches, num_frames, frame_size = frames.shape
+    frames = frames.flatten(end_dim=1)
+
+    # We're using a DataLoader to help with batching of frames
+    dataset = torch.utils.data.TensorDataset(frames)
     loader = torch.utils.data.DataLoader(
         dataset, batch_size=batch_size, shuffle=False, drop_last=False
     )
 
-    if center:
-        timestamps = (
-            torch.range(0, audio.shape[1], hop_size_samples) / input_sample_rate()
-        )
-    else:
-        # We will only use center=True in HEAR 2021
-        raise ValueError("center = False not supported")
-
     # Put the model into eval mode, and don't compute any gradients.
     model.eval()
     with torch.no_grad():
         # Iterate over all batches and accumulate the embeddings
-        # into allembs
-        allembs = defaultdict(list)
+        embeddings = defaultdict(list)
         for batch in loader:
-            # The dataset only has one element, which is the audio
-            # batch tensor
-            embs = model(batch[0], hop_size_samples=hop_size_samples, center=center)
-            for e in embs:
-                allembs[e].append(embs[e])
-    # Concatenate the minibatches before returning
-    # TODO: Check that returns are the right type?
-    for e in allembs:
-        allembs[e] = torch.cat(allembs[e], dim=0)
-        assert allembs[e].shape[0] == audio.shape[0]
-        assert len(timestamps) == allembs[e].shape[1]
-    return allembs, timestamps
+            result = model(batch[0])
+            for size, embedding in result.items():
+                embeddings[size].append(embedding)
+
+    # Concatenate mini-batches back together and unflatten the frames back
+    # to audio batches
+    embeddings = dict(embeddings)
+    for size, embedding in embeddings.items():
+        embeddings[size] = torch.cat(embedding, dim=0)
+        embeddings[size] = embeddings[size].unflatten(0, (audio_batches, num_frames))
```

Comment:
It's still very hard for me to tell if this is unframing the audio correctly for each sample. I suspect it's not correct but it's hard to tell just by inspection. I think we need some sort of test :\

Reply:
@turian What do you mean by "unframing the audio"?